packages/components/src/mobile/image/index.native.js

```diff
@@ -52,11 +54,15 @@ const ImageComponent = ( {
 	useEffect( () => {
 		if ( url ) {
 			Image.getSize( url, ( imgWidth, imgHeight ) => {
-				setImageData( {
+				const metaData = {
 					aspectRatio: imgWidth / imgHeight,
 					width: imgWidth,
 					height: imgHeight,
-				} );
+				};
+				setImageData( metaData );
+				if ( typeof onImageDataLoad === 'function' ) {
```

Comment:
We don't usually check the type, for this case just checking if it's not undefined is ok.

```suggestion
				if ( onImageDataLoad ) {
```